src/java.base/share/classes/java/nio/file/attribute/FileTime.java

```diff
@@ -232,36 +232,28 @@ public Instant toInstant() {
             long secs = 0L;
             int nanos = 0;
             switch (unit) {
-                case DAYS:
-                    secs = scale(value, SECONDS_PER_DAY,
-                                 Long.MAX_VALUE/SECONDS_PER_DAY);
-                    break;
-                case HOURS:
-                    secs = scale(value, SECONDS_PER_HOUR,
-                                 Long.MAX_VALUE/SECONDS_PER_HOUR);
-                    break;
-                case MINUTES:
-                    secs = scale(value, SECONDS_PER_MINUTE,
-                                 Long.MAX_VALUE/SECONDS_PER_MINUTE);
-                    break;
-                case SECONDS:
-                    secs = value;
-                    break;
-                case MILLISECONDS:
+                case DAYS    -> secs = scale(value, SECONDS_PER_DAY,
+                        Long.MAX_VALUE / SECONDS_PER_DAY);
+                case HOURS   -> secs = scale(value, SECONDS_PER_HOUR,
+                        Long.MAX_VALUE / SECONDS_PER_HOUR);
+                case MINUTES -> secs = scale(value, SECONDS_PER_MINUTE,
+                        Long.MAX_VALUE / SECONDS_PER_MINUTE);
+                case SECONDS -> secs = value;
+                case MILLISECONDS -> {
                     secs = Math.floorDiv(value, MILLIS_PER_SECOND);
-                    nanos = (int)Math.floorMod(value, MILLIS_PER_SECOND)
+                    nanos = (int) Math.floorMod(value, MILLIS_PER_SECOND)
                             * NANOS_PER_MILLI;
-                    break;
-                case MICROSECONDS:
+                }
+                case MICROSECONDS -> {
                     secs = Math.floorDiv(value, MICROS_PER_SECOND);
-                    nanos = (int)Math.floorMod(value, MICROS_PER_SECOND)
+                    nanos = (int) Math.floorMod(value, MICROS_PER_SECOND)
                             * NANOS_PER_MICRO;
-                    break;
-                case NANOSECONDS:
+                }
+                case NANOSECONDS  -> {
                     secs = Math.floorDiv(value, NANOS_PER_SECOND);
-                    nanos = (int)Math.floorMod(value, NANOS_PER_SECOND);
-                    break;
-                default : throw new AssertionError("Unit not handled");
+                    nanos = (int) Math.floorMod(value, NANOS_PER_SECOND);
+                }
+                default -> throw new AssertionError("Unit not handled");
```

Comment:
The formatting here is all messed up, I think you'll need to fix this up before it can be reviewed.

Reply:
Sorry about that. I've changed it now. See 2f179b5

Reply:
This still looks a bit messy because you've got 3 different styles in the one switch statement. It's okay to drop FileTime from the patch if you want as it's not worth spending time on.

Reply:
OK, if you prefer. I've reverted the changes now. See commit 23f53c5